packages/headless/src/plans.ts

```diff
@@ -31,3 +31,46 @@ export function useCreatePlan({
     onError,
   });
 }
+
+export function useUpdatePlan({
+  id,
+  onSuccess,
+  onError,
+}: {
+  id: number;
```

Comment:
The `id` of the plan should be passed to `mutation.mutate` an not to the hook. The hook should look like this. 

```ts
const updatePlan = useUpdatePlan({onSuccess, onError});
updatePlan.murate({id, payload})
```